### ISSUE TITLE
Generated migrations should inherit from versioned base class in Rails 5

### ISSUE THREAD

**Issue by derekprior** [thoughtbot/scenic#176]
Migrations are currently generated with the base class of `ActiveRecord::Migration`. If we're running under rails 5, we should generate them with `ActiveRecord::Migration[5.0]` to save our users from deprecation warnings.

It's not clear if we should hard code the version check and version number or if we should use something like:

```ruby
def migration_version
  if ActiveRecord::Migration.respond_to?("[]")
    "[#{Rails.version.major}.#{Rails.version.minor}]"
  end
end
```

I'm not sure if they intend to bump this with every version or what...

**Comment by calebthompson** [comment]
I'd be interested in seeing if we can get this automatically. @sgrif do you know if there's an API for this we could use?

**Comment by sgrif** [comment]
`ActiveRecord::Migration.current_version` is the public API to get the generated value.

**Comment by sgrif** [comment]
I think the best course of action would be to specifically target 5.0, and not the current version though.

**Comment by derekprior** [comment]
Why do you say that?

**Comment by sgrif** [comment]
Well, the point of the versioned migrations is that it makes us more free to change the migration API without breaking old migrations. You wrote the generator to generate migrations that are expected to work with the DSL as of 5.0. There's no guarantee that they'll continue to work with future versions, and there's no downside to being specific. (Yes, `execute` is unlikely to ever change, but in the more general sense...)

**Comment by teoljungberg** [comment]
I opened #179 to remedy this issue.

**Comment by teoljungberg** [comment]
This can be closed now since #179 is merged.
